src/engine/openmp.cc

```diff
@@ -74,17 +74,17 @@ void OpenMP::set_reserve_cores(int cores) {
   reserve_cores_ = cores;
 #ifdef _OPENMP
   if (reserve_cores_ >= omp_thread_max_) {
-    omp_set_num_threads(1);
+    omp_thread_max_ = 1;
   } else {
-    omp_set_num_threads(omp_thread_max_ - reserve_cores_);
+    omp_thread_max_ -= reserve_cores_;
   }
 #endif
 }
 
 int OpenMP::GetRecommendedOMPThreadCount(bool exclude_reserved) const {
 #ifdef _OPENMP
   if (omp_num_threads_set_in_environment_) {
-    return omp_get_max_threads();
+    return omp_thread_max_;
   }
   if (enabled_) {
     int thread_count = omp_get_max_threads();
```

Comment:
You mention you want to call `omp_get_max_threads()` once and cache the results. This is another place where the API is called (in addition to line 55). Is that intentional? will it not create the same issue?

Reply:
these are the lines I changed. I removed the call and replaced it with the cached result.

the only call is in the constructor (line 55).

Reply:
@azai91 Can you address @lupesko comments? I also found this change suspicious. It seems once omp_thread_max_ is set to 1 in line 77, all the subsequent call to this function will return 1 if the env variable is set. Which means, this stops multithreading altogether in the operator. Could you provide a test case to verify this change actually works? Thanks